test/index.ts

```diff
@@ -31,4 +33,59 @@ describe('Onboard Guilherme Brito', () => {
       expect(res.data.data.hello).to.be.equal('Hello World');
     });
   });
+
+  describe('#create user mutation', () => {
+    afterEach(async () => {
+      await prisma.user.deleteMany();
+    });
+
+    it('should create a user with the correct informations', async () => {
+      const userInput: UserInput = {
+        birthDate: '25/04/2003',
+        email: 'teste@taqtile.com.br',
+        name: 'guilherme',
+        password: 'senha123',
+      };
+
+      const graphqlMutation = `#graphql
+        mutation CreateUser($data: UserInput!) {
+          createUser(data: $data) {
+            birthDate
+            email
+            id
+            name
+          }
+        }`;
+
+      const graphqlMutationRequestBody = {
+        operationName: 'CreateUser',
+        query: graphqlMutation,
+        variables: { data: userInput },
+      };
+
+      const {
+        data: {
+          data: {
+            createUser: { name, email, birthDate },
+          },
+        },
+      } = await axios.post(LOCAL_SERVER_URL, graphqlMutationRequestBody);
```

Comment:
aqui eu não sei se é uma boa usar destructuring no createUser, poderia ter parado ali e comparado o resultado inteiro 🤔 
A motivação é que fazendo algo como `expect(createUser).to.be.deep.eq({ id: dbUser.id, name: 'guilherme', ... })` o teu teste vai quebrar, propositalmente, quando vc inserir outro campo ou ter alguma outra mudança estrutural, na abordagem atual só quebraria se vc tirasse name, email ou birthDate

Dai no exemplo que eu sugeri ja indica uma leve alteração na ordem dos expects, primeiro comparar o dbUser e ver se criou certo no banco pq dai vc ja reaproveita o id dele pra comparar a resposta do graphql

Reply:
Feito